exclude.txt

```diff
@@ -53,7 +53,6 @@ wp-content/mu-plugins/force-strong-passwords*
 wp-content/mu-plugins/wpengine-common*
 wp-content/mu-plugins/wpe-wp-sign-on-plugin*
 wp-content/mu-plugins/wpe-elasticpress-autosuggest-logger*
-wp-content/mu-plugins/wpe-cache-plugin/*
```

Comment:
This is odd, as the exclude list is to avoid rsync from operating on the directories and files matching the pattern, not delete it. 

Any chance you could add the build output (or at least the relevant part) to the PR? It would help troubleshoot the issue.

Reply:
```
Run wpengine/github-action-wpe-site-deploy@v3.0
  with:
    WPE_SSHG_KEY_PRIVATE: ***
  
    WPE_ENV: dev
    SRC_PATH: cms/
    FLAGS: -azvrh --delete --exclude=".htaccess, favicon.ico, apple-touch-icon.png, apple-touch-icon-precomposed.png"
    CACHE_CLEAR: true
    PHP_LINT: false
  env:
    ENV_NAME: DEV
    COMPOSER_ARGS: --prefer-dist
/usr/bin/docker run --name cd98f3c64733de6b04d9[2](https://github.com/Evolve-Design-Group/site/runs/7242437182?check_suite_focus=true#step:9:2)b4ed894f4270[3](https://github.com/Evolve-Design-Group/site/runs/7242437182?check_suite_focus=true#step:9:3)331_d6ba86 --label [4](https://github.com/Evolve-Design-Group/site/runs/7242437182?check_suite_focus=true#step:9:4)cd98f --workdir /github/workspace --rm -e ENV_NAME -e COMPOSER_ARGS -e INPUT_WPE_SSHG_KEY_PRIVATE -e INPUT_WPE_ENV -e INPUT_SRC_PATH -e INPUT_FLAGS -e INPUT_CACHE_CLEAR -e INPUT_PHP_LINT -e INPUT_REMOTE_PATH -e INPUT_PRD_ENV -e INPUT_STG_ENV -e INPUT_DEV_ENV -e INPUT_SCRIPT -e HOME -e GITHUB_JOB -e GITHUB_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_REPOSITORY_OWNER -e GITHUB_RUN_ID -e GITHUB_RUN_NUMBER -e GITHUB_RETENTION_DAYS -e GITHUB_RUN_ATTEMPT -e GITHUB_ACTOR -e GITHUB_WORKFLOW -e GITHUB_HEAD_REF -e GITHUB_BASE_REF -e GITHUB_EVENT_NAME -e GITHUB_SERVER_URL -e GITHUB_API_URL -e GITHUB_GRAPHQL_URL -e GITHUB_REF_NAME -e GITHUB_REF_PROTECTED -e GITHUB_REF_TYPE -e GITHUB_WORKSPACE -e GITHUB_ACTION -e GITHUB_EVENT_PATH -e GITHUB_ACTION_REPOSITORY -e GITHUB_ACTION_REF -e GITHUB_PATH -e GITHUB_ENV -e GITHUB_STEP_SUMMARY -e RUNNER_OS -e RUNNER_ARCH -e RUNNER_NAME -e RUNNER_TOOL_CACHE -e RUNNER_TEMP -e RUNNER_WORKSPACE -e ACTIONS_RUNTIME_URL -e ACTIONS_RUNTIME_TOKEN -e ACTIONS_CACHE_URL -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/_temp/_runner_file_commands":"/github/file_commands" -v "/home/runner/work/site/site":"/github/workspace" 4cd98f:3c64733de6b04d92b4ed894f42703331
Deploying refs/heads/develop to site ...
# site.ssh.wpengine.net:22 SSH-2.0-Go
prepping file perms...
file perms set...
Skipping PHP Linting.
OpenSSH_8.4p1, OpenSSL 1.1.1k  2[5](https://github.com/Evolve-Design-Group/site/runs/7242437182?check_suite_focus=true#step:9:5) Mar 2021
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: Authenticator provider $SSH_SK_PROVIDER did not resolve; disabling
debug1: Connecting to site.ssh.wpengine.net [35.237.148.8] port 22.
debug1: Connection established.
debug1: identity file /github/home/.ssh/github_action type 0
debug1: identity file /github/home/.ssh/github_action-cert type -1
debug1: Local version string SSH-2.0-OpenSSH_8.4
debug1: Remote protocol version 2.0, remote software version Go
debug1: no match: Go
debug1: Authenticating to site.ssh.wpengine.net:22 as 'wpe_gha+site'
debug1: SSH2_MSG_KEXINIT sent
debug1: SSH2_MSG_KEXINIT received
debug1: kex: algorithm: curve25519-sha25[6](https://github.com/Evolve-Design-Group/site/runs/7242437182?check_suite_focus=true#step:9:6)@libssh.org
debug1: kex: host key algorithm: rsa-sha2-512
debug1: kex: server->client cipher: aes128-ctr MAC: hmac-sha2-256-etm@openssh.com compression: none
debug1: kex: client->server cipher: aes128-ctr MAC: hmac-sha2-256-etm@openssh.com compression: none
debug1: expecting SSH2_MSG_KEX_ECDH_REPLY
debug1: Server host key: ssh-rsa SHA256:T8IoIgBRJ3gDjA6/q[7](https://github.com/Evolve-Design-Group/site/runs/7242437182?check_suite_focus=true#step:9:7)i33+nSrrpVfZipYtxo1rqvww4
Warning: Permanently added 'site.ssh.wpengine.net,35.237.14[8](https://github.com/Evolve-Design-Group/site/runs/7242437182?check_suite_focus=true#step:9:8).8' (RSA) to the list of known hosts.
debug1: rekey out after 42[9](https://github.com/Evolve-Design-Group/site/runs/7242437182?check_suite_focus=true#step:9:9)4967296 blocks
debug1: SSH2_MSG_NEWKEYS sent
debug1: expecting SSH2_MSG_NEWKEYS
debug1: SSH2_MSG_NEWKEYS received
debug1: rekey in after 4294967296 blocks
debug1: Will attempt key: /github/home/.ssh/github_action RSA SHA256:TN+ML/fUGJAD2CPIBdtQ[11](https://github.com/Evolve-Design-Group/site/runs/7242437182?check_suite_focus=true#step:9:11)pj41pqhHljzAetYB5/0FI explicit
debug1: SSH2_MSG_SERVICE_ACCEPT received
debug1: Authentications that can continue: publickey
debug1: Next authentication method: publickey
debug1: Offering public key: /github/home/.ssh/github_action RSA SHA256:TN+ML/fUGJAD2CPIBdtQ11pj41pqhHljzAetYB5/0FI explicit
debug1: Server accepts key: /github/home/.ssh/github_action RSA SHA256:TN+ML/fUGJAD2CPIBdtQ11pj41pqhHljzAetYB5/0FI explicit
debug1: Authentication succeeded (publickey).
!!! MASTER SSH CONNECTION ESTABLISHED !!!
Authenticated to site.ssh.wpengine.net ([35.237.148.8]:22).
debug1: setting up multiplex master socket
debug1: channel 0: new [/github/home/.ssh/ctl/26d97fdf401a8e5a44fe94[12](https://github.com/Evolve-Design-Group/site/runs/7242437182?check_suite_focus=true#step:9:12)dac6b1fef4ae80bd]
debug1: forking to background
debug1: Entering interactive session.
debug1: pledge: id
OpenSSH_8.4p1, OpenSSL 1.1.1k  25 Mar 2021
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: Authenticator provider $SSH_SK_PROVIDER did not resolve; disabling
debug1: multiplexing control connection
debug1: channel 1: new [mux-control]
debug1: channel 2: new [client-session]
debug1: Sending command: rsync --server -vlogDtprze.iLsfxCIvu --delete . sites/site/
debug1: mux_client_request_session: master session id: 2
rsync: [sender] link_stat "/github/workspace/favicon.ico," failed: No such file or directory (2)
rsync: [sender] link_stat "/github/workspace/apple-touch-icon.png," failed: No such file or directory (2)
rsync: [sender] link_stat "/github/workspace/apple-touch-icon-precomposed.png"" failed: No such file or directory (2)
sending incremental file list
deleting favicon.ico
deleting apple-touch-icon.png
deleting apple-touch-icon-precomposed.png
deleting .htaccess
./
readme.html
wp-cron.php
wp-links-opml.php
wp-load.php
wp-login.php
wp-mail.php
wp-settings.php
wp-signup.php
wp-trackback.php
deleting wp-content/mu-plugins/wpe-cache-plugin/view/wpe-cache-page.php
deleting wp-content/mu-plugins/wpe-cache-plugin/view/quick-actions.php
deleting wp-content/mu-plugins/wpe-cache-plugin/view/cache-times.php
deleting wp-content/mu-plugins/wpe-cache-plugin/view/
deleting wp-content/mu-plugins/wpe-cache-plugin/security/security-checks.php
deleting wp-content/mu-plugins/wpe-cache-plugin/security/
deleting wp-content/mu-plugins/wpe-cache-plugin/js/utils/Time.js
deleting wp-content/mu-plugins/wpe-cache-plugin/js/utils/DateTime.js
deleting wp-content/mu-plugins/wpe-cache-plugin/js/utils/
deleting wp-content/mu-plugins/wpe-cache-plugin/js/services/CachePluginWindowModifier.js
deleting wp-content/mu-plugins/wpe-cache-plugin/js/services/CachePluginQueryParams.js
deleting wp-content/mu-plugins/wpe-cache-plugin/js/services/CachePluginApiService.js
deleting wp-content/mu-plugins/wpe-cache-plugin/js/services/
deleting wp-content/mu-plugins/wpe-cache-plugin/js/dist/wpe-cache-plugin-admin.js.map
deleting wp-content/mu-plugins/wpe-cache-plugin/js/dist/wpe-cache-plugin-admin.js
deleting wp-content/mu-plugins/wpe-cache-plugin/js/dist/
deleting wp-content/mu-plugins/wpe-cache-plugin/js/components/LastErrorText.js
deleting wp-content/mu-plugins/wpe-cache-plugin/js/components/LastClearedText.js
deleting wp-content/mu-plugins/wpe-cache-plugin/js/components/JQTextElement.js
deleting wp-content/mu-plugins/wpe-cache-plugin/js/components/JQElement.js
deleting wp-content/mu-plugins/wpe-cache-plugin/js/components/ErrorToast.js
deleting wp-content/mu-plugins/wpe-cache-plugin/js/components/ClearAllCacheIcon.js
deleting wp-content/mu-plugins/wpe-cache-plugin/js/components/ClearAllCacheBtn.js
deleting wp-content/mu-plugins/wpe-cache-plugin/js/components/CacheTimesFormReferField.js
deleting wp-content/mu-plugins/wpe-cache-plugin/js/components/
deleting wp-content/mu-plugins/wpe-cache-plugin/js/index.js
deleting wp-content/mu-plugins/wpe-cache-plugin/js/
deleting wp-content/mu-plugins/wpe-cache-plugin/css/cache-styles.css
deleting wp-content/mu-plugins/wpe-cache-plugin/css/
deleting wp-content/mu-plugins/wpe-cache-plugin/wpe-common-adapter.php
deleting wp-content/mu-plugins/wpe-cache-plugin/wpe-admin.php
deleting wp-content/mu-plugins/wpe-cache-plugin/register-rest-endpoints.php
deleting wp-content/mu-plugins/wpe-cache-plugin/plugin-rest-paths.php
deleting wp-content/mu-plugins/wpe-cache-plugin/max-cdn-provider.php
deleting wp-content/mu-plugins/wpe-cache-plugin/logging-trait.php
deleting wp-content/mu-plugins/wpe-cache-plugin/date-time-helper.php
deleting wp-content/mu-plugins/wpe-cache-plugin/clear-all-caches-controller.php
deleting wp-content/mu-plugins/wpe-cache-plugin/clear-all-cache-status.php
deleting wp-content/mu-plugins/wpe-cache-plugin/cache-view-helpers.php
deleting wp-content/mu-plugins/wpe-cache-plugin/cache-setting-values.php
deleting wp-content/mu-plugins/wpe-cache-plugin/cache-db-settings.php
deleting wp-content/mu-plugins/wpe-cache-plugin/cache-control.php
deleting wp-content/mu-plugins/wpe-cache-plugin/
deleting wp-content/mu-plugins/wpe-cache-plugin.php
cannot delete non-empty directory: wp-content/mu-plugins
# All other files go here
debug1: client_input_channel_req: channel 2 rtype exit-status reply 0
debug1: channel 2: free: client-session, nchannels 3
debug1: channel 1: free: mux-control, nchannels 2
rsync error: some files/attrs were not transferred (see previous errors) (code 23) at main.c(1330) [sender=v3.13.0_rc2-264-g725ac7fb]
sent 25.97M bytes  received 447.12K bytes  838.53K bytes/sec
total size is 134.90M  speedup is 5.11
```

Reply:
Thanks for posting the output!

I agree that we should remove this line because it's redundant, but should point out that it doesn't fix any conflicts with `wpe-cache-plugin`. The conflicts you're experiencing in `v3.0` are resolved by #8, but that change only exists on `main` and has not yet been released. Since this feature branch is based off of `main` rather than the `v3.0` tag, it already includes the #8 fix and will appear as though it addresses the conflict no matter what additional changes are introduced. Again, this doesn't affect the change you've made here. I just wanted to capture that context in case we ever need to refer back to it 🙂